core/slice.go

```diff
@@ -690,6 +711,63 @@ func (sl *Slice) SendPendingEtxsToDom(pEtxs types.PendingEtxs) error {
 	return sl.domClient.SendPendingEtxsToDom(context.Background(), pEtxs)
 }
 
+func (sl *Slice) GetPEtxRollupAfterRetryThreshold(blockHash common.Hash, hash common.Hash, location common.Location) (types.PendingEtxsRollup, error) {
+	pEtx, exists := sl.pEtxRetryCache.Get(blockHash)
+	if !exists || pEtx.(pEtxRetry).retries < c_pEtxRetryThreshold {
+		return types.PendingEtxsRollup{}, ErrPendingEtxNotFound
+	}
+	return sl.GetPendingEtxsRollupFromSub(hash, location)
+}
+
+// GetPendingEtxsRollupFromSub gets the pending etxs rollup from the appropriate prime
+func (sl *Slice) GetPendingEtxsRollupFromSub(hash common.Hash, location common.Location) (types.PendingEtxsRollup, error) {
+	nodeCtx := common.NodeLocation.Context()
+	if nodeCtx == common.PRIME_CTX {
+		if sl.subClients[location.SubIndex()] != nil {
+			pEtxRollup, err := sl.subClients[location.SubIndex()].GetPendingEtxsRollupFromSub(context.Background(), hash, location)
+			if err != nil {
+				return types.PendingEtxsRollup{}, err
+			} else {
+				sl.AddPendingEtxsRollup(pEtxRollup)
```

Comment:
There's no return here so the function will return an error. Is that intentional?